src/main/java/io/cdap/plugin/servicenow/sink/service/ServiceNowSinkAPIRequestImpl.java

```diff
@@ -0,0 +1,251 @@
+      restRequest.setUrl(String.format(ServiceNowConstants.UPDATE_TABLE_API_URL_TEMPLATE, config.getTableName(),
+        sysId));
+    }
+    restRequest.setId(counter.toString());
+    // Increment Rest Request counter
+    counter++;
+    restRequest.setHeaders(headers);
+    restRequest.setMethod(ServiceNowConstants.HTTP_POST);
+    if (config.getOperation().equals(ServiceNowConstants.UPDATE_OPERATION)) {
+      restRequest.setMethod(ServiceNowConstants.HTTP_PUT);
+    }
+    restRequest.setBody(encodedData);
+    return restRequest;
+  }
+
+  /**
+   * Inserts/Updates the list of records into ServiceNow table
+   *
+   * @param restRequestList The list of restRequests
+   * @param records The list of records to insert/update
+   * @param accessToken The access token
+   * @return true if the apiResponse code is 200 otherwise false
+   */
+  public Boolean createPostRequest(List<RestRequest> restRequestList, List<RestRequest> records, String accessToken)
+    throws OAuthProblemException, InterruptedException {
+    ServiceNowBatchRequest payloadRequest = getPayloadRequest(records);
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint());
+    RestAPIResponse apiResponse;
+
+    try {
+      requestBuilder.setAuthHeader(accessToken);
+      requestBuilder.setAcceptHeader(MediaType.APPLICATION_JSON);
+      requestBuilder.setContentTypeHeader(MediaType.APPLICATION_JSON);
+      StringEntity stringEntity = new StringEntity(gson.toJson(payloadRequest), ContentType.APPLICATION_JSON);
+      requestBuilder.setEntity(stringEntity);
+      apiResponse = restApi.executePost(requestBuilder.build());
+
+      if (!apiResponse.isSuccess()) {
+        LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
+        throw new RetryableException();
+      } else {
+        JsonObject responseJSON = jsonParser.parse(apiResponse.getResponseBody()).getAsJsonObject();
+        JsonArray servicedRequestsArray = responseJSON.get(ServiceNowConstants.SERVICED_REQUESTS).getAsJsonArray();
+        for (int i = 0; i < servicedRequestsArray.size(); i++) {
+          if (servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.STATUS_CODE)
+            .getAsInt() == HttpStatus.SC_FORBIDDEN) {
+            String encodedResponseBody = servicedRequestsArray.get(i).getAsJsonObject().get(ServiceNowConstants.BODY)
+              .getAsString();
+            String decodedResponseBody = new String(Base64.getDecoder().decode(encodedResponseBody));
+            String errorDetail = jsonParser.parse(decodedResponseBody).getAsJsonObject().get(ServiceNowConstants.ERROR)
+              .getAsJsonObject().get(ServiceNowConstants.ERROR_DETAIL).getAsString();
+            if (errorDetail.equals(ServiceNowConstants.ACL_EXCEPTION)) {
+              throw new RuntimeException(String.format("Permission denied for '%s' operation.", config.getOperation()));
```

Comment:
are we still using RuntimeException here? Wouldn't IOException be a better way.

Reply:
Changed to `IllegalStateException` as `IOException` is being retried at line no. 173. 659b6bd2dc503d409443ef1b43804566a4e25be0